tracer/contrib/go-redis/tracedredis.go

```diff
@@ -0,0 +1,112 @@
+package tracedredis
+
+import (
+	"context"
+	"github.com/DataDog/dd-trace-go/tracer"
+	"github.com/DataDog/dd-trace-go/tracer/ext"
+	"github.com/go-redis/redis"
+	"strconv"
+	"strings"
+)
+
+func NewTracedClient(opt *redis.Options, t *tracer.Tracer, service string) *TracedClient {
+	var host, port string
+	addr := strings.Split(opt.Addr, ":")
+	host = addr[0]
+	port = addr[1]
+	db := strconv.Itoa(opt.DB)
+
+	client := redis.NewClient(opt)
+	t.SetServiceInfo(service, "redis", ext.AppTypeDB)
+	tc := &TracedClient{
+		client,
+		host,
+		port,
+		db,
+		service,
+		t,
+	}
+
+	tc.Client.WrapProcess(createWrapperFromClient(tc))
+	return tc
+}
+
+type TracedClient struct {
+	*redis.Client
+	host    string
+	port    string
+	db      string
+	service string
+	tracer  *tracer.Tracer
+}
+
+func (c *TracedClient) SetContext(ctx context.Context) {
+	c.Client = c.Client.WithContext(ctx)
+}
+
+type TracedPipeline struct {
+	*redis.Pipeline
+	host    string
+	port    string
+	db      string
+	service string
+	tracer  *tracer.Tracer
+}
+
+func (c *TracedClient) Pipeline() *TracedPipeline {
+	return &TracedPipeline{
+		c.Client.Pipeline(),
+		c.host,
+		c.port,
+		c.db,
+		c.service,
+		c.tracer,
+	}
+}
+
+func (c *TracedPipeline) TracedExec(ctx context.Context) ([]redis.Cmder, error) {
```

Comment:
the equivalent from database sql is `ExecContext`

https://golang.org/pkg/database/sql/#DB.ExecContext